pkg/config/editor_presets.go

```diff
@@ -0,0 +1,113 @@
+	preset := getPreset(osConfig, guessDefaultEditor)
+	template := osConfig.Edit
+	if template == "" {
+		template = preset.editTemplate
+	}
+
+	return template, getEditInTerminal(osConfig, preset)
+}
+
+func GetEditAtLineTemplate(osConfig *OSConfig, guessDefaultEditor func() string) (string, bool) {
+	preset := getPreset(osConfig, guessDefaultEditor)
+	template := osConfig.EditAtLine
+	if template == "" {
+		template = preset.editAtLineTemplate
+	}
+	return template, getEditInTerminal(osConfig, preset)
+}
+
+func GetEditAtLineAndWaitTemplate(osConfig *OSConfig, guessDefaultEditor func() string) string {
+	preset := getPreset(osConfig, guessDefaultEditor)
+	template := osConfig.EditAtLineAndWait
+	if template == "" {
+		template = preset.editAtLineAndWaitTemplate
+	}
+	return template
+}
+
+type editPreset struct {
+	editTemplate              string
+	editAtLineTemplate        string
+	editAtLineAndWaitTemplate string
+	editInTerminal            bool
+}
+
+func getPreset(osConfig *OSConfig, guessDefaultEditor func() string) *editPreset {
+	presets := map[string]*editPreset{
+		"vim": {
+			editTemplate:              "vim -- {{filename}}",
+			editAtLineTemplate:        "vim +{{line}} -- {{filename}}",
+			editAtLineAndWaitTemplate: "vim +{{line}} -- {{filename}}",
+			editInTerminal:            true,
+		},
+		"emacs": {
+			editTemplate:              "emacs -- {{filename}}",
+			editAtLineTemplate:        "emacs +{{line}} -- {{filename}}",
+			editAtLineAndWaitTemplate: "emacs +{{line}} -- {{filename}}",
+			editInTerminal:            true,
+		},
+		"nano": {
+			editTemplate:              "nano -- {{filename}}",
+			editAtLineTemplate:        "nano +{{line}} -- {{filename}}",
+			editAtLineAndWaitTemplate: "nano +{{line}} -- {{filename}}",
+			editInTerminal:            true,
+		},
+		"vscode": {
+			editTemplate:              "code --reuse-window -- {{filename}}",
+			editAtLineTemplate:        "code --reuse-window --goto -- {{filename}}:{{line}}",
+			editAtLineAndWaitTemplate: "code --reuse-window --goto --wait -- {{filename}}:{{line}}",
+			editInTerminal:            false,
+		},
+		"sublime": {
+			editTemplate:              "subl -- {{filename}}",
+			editAtLineTemplate:        "subl -- {{filename}}:{{line}}",
+			editAtLineAndWaitTemplate: "subl --wait -- {{filename}}:{{line}}",
+			editInTerminal:            false,
+		},
+		"bbedit": {
+			editTemplate:              "bbedit -- {{filename}}",
+			editAtLineTemplate:        "bbedit +{{line}} -- {{filename}}",
+			editAtLineAndWaitTemplate: "bbedit +{{line}} --wait -- {{filename}}",
+			editInTerminal:            false,
+		},
+		"xcode": {
+			editTemplate:              "xed -- {{filename}}",
+			editAtLineTemplate:        "xed --line {{line}} -- {{filename}}",
+			editAtLineAndWaitTemplate: "xed --line {{line}} --wait -- {{filename}}",
+			editInTerminal:            false,
+		},
+	}
+
+	editorToPreset := map[string]string{
+		"vim":    "vim",
```

Comment:
we should also add `nvim` and `vi` as those were previously covered in `GetEditCmdStr`

Reply:
This took three fixup commits :) I took the opportunity to simplify the code a bit, see 1ba3f1018e38, f5c1a0561ce2, and 7685a04aeec9.